wordpress-mobile/WordPress-iOS
# Add media filters

This is something that was originally suggested in the Modernization spec, but I didn't have time to implement, so here it is now.

- Add media filters
- Add “Aspect Fill Grid” support to iPhone

Not in the scope: changing the empty state; seems acceptable.

To test:

- Verify that the navigation title now has a menu that allows you to switch between media types and also change the display mode: aspect/square
- Verify that the aspect ratio display mode is now available on iPhone
- Verify that the selection is **not** persisted
- Verify that when you make the selection, the navigation title also updates 

<img width="340" alt="Screenshot 2023-11-27 at 8 55 03 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/c6d7c8ec-d96e-4d0e-8577-fef5cdaae4d1"> <img width="340" alt="Screenshot 2023-11-27 at 8 55 08 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/3a74005b-5756-4b40-8a2d-113ba5ad4e70">

<img width="1316" alt="Screenshot 2023-11-27 at 8 57 40 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/f5f942d1-c66e-4e54-910f-54ce49698f88">

## Regression Notes
1. Potential unintended areas of impact: Site Media
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:


UI Changes testing checklist:
